infrastructure/simulation-tool/src/utils/number.utils.ts

```diff
@@ -0,0 +1,24 @@
+import { BigNumber, BigNumberish } from 'ethers';
+
+type RandomAmountGenerator = Generator<BigNumber, BigNumber>;
+
+const getRandomBigNumber = (min: BigNumberish, max: BigNumberish): BigNumber => {
+    const diff = BigNumber.from(max).sub(min);
+    const randomDecimal = Math.random();
+    const scaled = diff.mul(BigNumber.from((randomDecimal * 1e18).toFixed()));
+    const randomNumber = scaled.div(BigNumber.from((1e18).toString()));
+
+    return randomNumber.add(min);
+};
+
+function* createRandomAmountGenerator(minMaxTuple: [BigNumberish, BigNumberish]): RandomAmountGenerator {
```

Comment:
Im assuming that is for future implementations, because I don't see any usage in the code.

Reply:
well, initially I was creating this generator per key in weiLimits in the setup stage. I have removed that, though I thought I might do something similar in the future refactor, so I kept the function.